internal/mockgen/runner.go

```diff
@@ -18,27 +20,44 @@ type Runner interface {
 
 func Checksum(r Runner) (string, error) {
 	d := r.GetDestination()
-	tmpFile := util.TmpFilePath(d)
-	defer os.Remove(tmpFile)
+	tmpFilePath := util.TmpFilePath(d)
+	defer os.Remove(tmpFilePath)
 
 	// use tmpfile to test generating mock
-	r.SetDestination(tmpFile)
+	r.SetDestination(tmpFilePath)
 	defer r.SetDestination(d)
 
 	if err := r.Run(); err != nil {
 		return "", fmt.Errorf("failed to run mockgen: %v \nPlease run `%s` and check if mockgen works correctly with your options", err, r)
 	}
 
-	checksum, err := util.CalculateCheckSum(tmpFile)
+	tmpFile, err := ioutil.ReadFile(tmpFilePath)
+	if err != nil {
+		return "", fmt.Errorf("failed read file. filename: %s, err: %w", tmpFilePath, err)
+	}
+
+	// See https://github.com/sanposhiho/gomockhandler/issues/88
+	adjustedFile := replaceTmpPathWithOriginal(tmpFilePath, tmpFile)
+
+	checksum, err := util.CalculateCheckSum([]byte(adjustedFile))
 	if err != nil {
 		return "", fmt.Errorf("calculate checksum of the mock: %v", err)
 	}
 
 	return checksum, nil
 }
 
+func replaceTmpPathWithOriginal(tmpFilePath string, tmpFile []byte) string {
+	originFilePath := strings.Replace(tmpFilePath, "tmp_", "", 1)
+	return strings.Replace(string(tmpFile), tmpFilePath, originFilePath, 1)
+}
```

Comment:
We should take care of a (rare) scenario where the original path contains `tmp_`.

```suggestion
func replaceTmpPathWithOriginal(tmpFilePath string, tmpFile []byte) string {
	d, f := filepath.Split(tmpFilePath)
	originFilePath := filepath.Join(d, strings.Replace(f, "tmp_", "", 1))
	return strings.Replace(string(tmpFile), tmpFilePath, originFilePath, 1)
}
```

Reply:
e.g., the current version doesn't work if the original file path is:
- "/hoge/tmp_fuga/file.go"
- "/hoge/fuga/tmp_file.go"

Reply:
I fixed and confirmed the operation verification again.